### PR TITLE
docs: document single-node upgrade support

### PR DESCRIPTION
**Pull request by sinnykumari**

### CODE REVIEW

File: docs/SingleNodeOpenShift.md [hunk 1]
```diff
@@ -24,8 +24,18 @@ With the exception of [optimized updates](./MachineConfigDaemon.md#optimized-upd
...
+
+## Post reboot cluster safety
+Any configuration applied using MCO on node including upgrades requires node reboot with the exception of [optimized updates](./MachineConfigDaemon.md#optimized-updates). MCO doesn't provide guarantee for post-reboot cluster safety. This implies that MCO will try its best to keep feature parity and behavior on SNO similar to regular OCP cluster. But due to the nature of SNO (single control plane node), when control plane node reboots during update/upgrade, node may not come up or join cluster due to multiple reasons that is not related to MCO, such as:
+
+- Bug in one of the package that was updated
+- Faulty config applied on node by user
+- Manual ssh into node by user and performing config change
+- Bug in another OCP component causing cluster not being fully available
+
+As a result, it is not possible for MCO team to know what caused the issue in the cluster post reboot unless there is a clear way to demonstrate (like easily reproducible) that cause of the problem is MCO.
```
Comment (yuqi-zhang) [inline docs/SingleNodeOpenShift.md:41]: Maybe rephrase as: "In these scenarios it is harder to find the exact cause of a problematic node post-reboot, and as a consequence it is harder to debug and recover from these cases. For the MCO to help debug, we would need easily reproducible steps, or system logging that points to MCO being the cause of the issue".

**Review by yuqi-zhang — APPROVED** [review]
overall lgtm, one minor suggestion

File: docs/SingleNodeOpenShift.md (hunk 1 above)
Comment (sinnykumari) [reply docs/SingleNodeOpenShift.md]: updated text with your rewording :)

**Comment by sinnykumari** [comment]
can I get lgtm label.

**Comment by yuqi-zhang** [comment]
/lgtm

Sorry for the delay
